modules/mqtt/mqtt-options.c

```diff
@@ -0,0 +1,310 @@
+{
+  g_free(self->ca_dir);
+  self->ca_dir = g_strdup(ca_dir);
+}
+
+void
+mqtt_client_options_set_ca_file(MQTTClientOptions *self, const gchar *ca_file)
+{
+  g_free(self->ca_file);
+  self->ca_file = g_strdup(ca_file);
+}
+
+void
+mqtt_client_options_set_cert_file(MQTTClientOptions *self, const gchar *cert_file)
+{
+  g_free(self->cert_file);
+  self->cert_file = g_strdup(cert_file);
+}
+
+void
+mqtt_client_options_set_key_file(MQTTClientOptions *self, const gchar *key_file)
+{
+  g_free(self->key_file);
+  self->key_file = g_strdup(key_file);
+}
+
+void
+mqtt_client_options_set_cipher_suite(MQTTClientOptions *self, const gchar *ciphers)
+{
+  g_free(self->ciphers);
+  self->ciphers = g_strdup(ciphers);
+}
+
+gboolean
+mqtt_client_options_set_ssl_version(MQTTClientOptions *self, const gchar *value)
+{
+  if (strcasecmp(value, "default") == 0)
+    self->ssl_version = MQTT_SSL_VERSION_DEFAULT;
+  else if (strcasecmp(value, "tlsv1_0") == 0)
+    self->ssl_version = MQTT_SSL_VERSION_TLS_1_0;
+  else if (strcasecmp(value, "tlsv1_1") == 0)
+    self->ssl_version = MQTT_SSL_VERSION_TLS_1_1;
+  else if (strcasecmp(value, "tlsv1_2") == 0)
+    self->ssl_version = MQTT_SSL_VERSION_TLS_1_2;
+  else
+    return FALSE;
+
+  return TRUE;
+}
+
+void
+mqtt_client_options_set_peer_verify(MQTTClientOptions *self, gboolean verify)
+{
+  self->peer_verify = verify;
+}
+
+void
+mqtt_client_options_use_system_cert_store(MQTTClientOptions *self, gboolean use_system_cert_store)
+{
+  /* TODO: auto_detect_ca_file() from the HTTP module */
+  self->use_system_cert_store = use_system_cert_store;
+}
+
+void
+mqtt_client_options_set_log_ssl_error_fn(MQTTClientOptions *self, gpointer context, gint(*log_error)(const gchar *str,
+                                         gsize len, gpointer u))
+{
+  self->context = context;
+  self->log_error = log_error;
+}
+
+gchar *
+mqtt_client_options_get_address(MQTTClientOptions *self)
+{
+  return self->address;
+}
+
+gint
+mqtt_client_options_get_qos(MQTTClientOptions *self)
+{
+  return self->qos;
+}
+
+gchar *
+mqtt_client_options_get_client_id(MQTTClientOptions *self)
+{
+  return self->client_id;
+}
+
+static void
+_set_ssl_options(MQTTClientOptions *option, MQTTClient_SSLOptions *ssl_opts)
+{
+  *ssl_opts = (MQTTClient_SSLOptions) MQTTClient_SSLOptions_initializer;
+  ssl_opts->trustStore = option->ca_file;
+  ssl_opts->CApath = option->ca_dir;
+  ssl_opts->keyStore = option->cert_file;
+  ssl_opts->privateKey = option->key_file;
+  ssl_opts->enabledCipherSuites = option->ciphers;
+  ssl_opts->sslVersion = option->ssl_version;
+  ssl_opts->enableServerCertAuth = option->peer_verify;
+  ssl_opts->verify = option->peer_verify;
+  ssl_opts->disableDefaultTrustStore = !option->use_system_cert_store;
+  ssl_opts->ssl_error_cb = option->log_error;
+  ssl_opts->ssl_error_context = option->context;
+}
+
+void
+mqtt_client_options_to_mqtt_client_connection_option(MQTTClientOptions *option, MQTTClient_connectOptions *conn_opts,
```

Comment:
Please rename `option` to `self`.

Reply:
commit: 728c8d708b867f760019fdc2f3461d4b1fe12976